kyaml/runfn/runfn.go

```diff
@@ -169,8 +175,29 @@ func (r RunFns) runFunctions(
 		// the output is nil (reading from Input)
 		outputs = append(outputs, kio.ByteWriter{Writer: r.Output})
 	}
-	err := kio.Pipeline{
-		Inputs: []kio.Reader{input}, Filters: fltrs, Outputs: outputs}.Execute()
+
+	var err error
+	pipeline := kio.Pipeline{
+		Inputs: []kio.Reader{input},
+		Filters: fltrs,
+		Outputs: outputs,
+	}
+	if r.LogSteps {
+		err = pipeline.ExecuteWithCallback(func (op kio.Filter) {
+			var identifier string
+			if execFilter, ok := op.(*exec.Filter); ok {
```

Comment:
You can use something like this

```
switch reflect.TypeOf(op){
  case *exec.Filter:
  case *container.Filter:
   ...
  default:
       ...
}
```

Reply:
So generally, are you ok with simply printing out unknown-type Filter as strings?

Reply:
I don't think printing out it string is a good idea since the Filter many contain quite large data or is not easy to read as strings. Actually the default case shouldn't be touched unless in the tests. So the default case can print some error.

Reply:
How about printing `Running unknown-type function` in the default case?

Reply:
OK. I'm fine with it.